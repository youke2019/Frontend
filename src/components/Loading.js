import React from "react";
import {
    ActivityIndicator,
    Modal,
    Text,
    StyleSheet,
    View,
} from "react-native";

class Loading extends React.Component {
    render(){
        const {visible = true} = this.props;
        return (
            <Modal
                visible={visible}
                transparent={true}
            >
                <View style={styles.center}>
                    <View style={styles.box}>
                        <ActivityIndicator
                            size="large"
                            color="white"
                        />
                        <Text style={styles.text}>加载中</Text>
                    </View>
                </View>
            </Modal>
        )
    }
}

const styles = StyleSheet.create({
    center:{
        flex:1,
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: 'transparent'
    },
    box: {
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: 'black',
        opacity: 0.6,
        height: 100,
        width: 100
    },
    text: {
        padding: 6,
        color: 'white',
        textAlign: 'center',
    }
});

export default Loading