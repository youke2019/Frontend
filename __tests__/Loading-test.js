import React from 'react';
import Loading from '../src/components/Loading';

import renderer from 'react-test-renderer';

test('renders home page correctly', () => {
    const tree = renderer.create(<Loading />).toJSON();
    expect(tree).toMatchSnapshot();
});