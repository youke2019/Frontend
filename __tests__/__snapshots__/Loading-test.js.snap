// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home page correctly 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "height": 100,
          "justifyContent": "center",
          "width": 100,
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="yellow"
        hidesWhenStopped={true}
        size="large"
      />
      <Text
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        加载中
      </Text>
    </View>
  </View>
</Modal>
`;
